pkg/controller/v1/xgboostjob/xgboostjob_controller.go

```diff
@@ -113,26 +112,29 @@ func newReconciler(mgr manager.Manager) reconcile.Reconciler {
 	}
 
 	// Create clients.
-	kubeClientSet, _, kubeBatchClientSet, err := createClientSets(kcfg)
+	kubeClientSet, _, volcanoClientSet, err := createClientSets(kcfg)
 	if err != nil {
 		log.Info("Error building kubeclientset: %s", err.Error())
 	}
 
+	// Create Informer factory
 	xgboostjob := &v1xgboost.XGBoostJob{}
 
 	gangScheduling := isGangSchedulerSet(xgboostjob.Spec.XGBReplicaSpecs)
 
 	log.Info("gang scheduling is set: ", "gangscheduling", gangScheduling)
 
 	// Initialize common job controller with components we only need.
-	r.xgbJobController = job_controller.JobController{
-		Controller:         r,
-		Expectations:       k8scontroller.NewControllerExpectations(),
-		Config:             v1.JobControllerConfiguration{EnableGangScheduling: gangScheduling},
-		WorkQueue:          &FakeWorkQueue{},
-		Recorder:           r.recorder,
-		KubeClientSet:      kubeClientSet,
-		KubeBatchClientSet: kubeBatchClientSet,
+	r.JobController = common.JobController{
+		Controller:       r,
+		Expectations:     expectation.NewControllerExpectations(),
+		Config:           common.JobControllerConfiguration{EnableGangScheduling: gangScheduling},
+		WorkQueue:        &FakeWorkQueue{},
+		Recorder:         r.recorder,
+		KubeClientSet:    kubeClientSet,
+		VolcanoClientSet: volcanoClientSet,
```

Comment:
for common, volcano is the default scheduler?

Reply:
@merlintang Yes. Most of the kube-batch contributors come to volcano project now. This is a kind of placeholder and make sure it's common compatible. In the previous implementation, we import `kube-batch` as well but not use it. 


xgboost itself can still be installed and used separately.  This is transparent to users and user don't need to use it. I should add more comments on the code. 

One improvement we can do here is to hide these details to different operators. I will add one issue https://github.com/kubeflow/common/issues/96